File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/repository/Repository.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.repository
{
    public interface Repository<T, Tid>
    {
        int add(T item);
        void delete(T item);
        void update(T item, Tid id);
        T? findById(Tid id);
        List<T> findAll();
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/repositories/Repository.java
package root.proiect_mpp.repositories;

import java.util.Collection;

public interface Repository<T, Tid> {
    int add(T elem);
    void delete(T elem);
    void update(T elem, Tid id);
    T findById(Tid id);
    Collection<T> getAll();
}
